### ISSUE TITLE
Excluding rules in config not working

### ISSUE THREAD

**Issue by dikmax** [dart-lang/linter#148]
I've added to pubspec.yaml linter options

```yaml
linter:
  rules:
    style_guide:
      implementation_imports: false
```

Then trying to run linter:

```bash
 pub run linter --config pubspec.yaml lib/generators/entities_generator.dart
```

It still shows error "Don't import implementation files from another package."

**Comment by zoechi** [comment]
How do you figure this should be possible this way, never saw it mentioned it should work this way.

The readme in https://pub.dartlang.org/packages/analyzer explains how to configure the analyzer to check specific lint rules by adding an `.analysis_options` file to your project (in the directory where the `pubspec.yaml` file is and listing the rules you want to have checked like 

```yaml
analyzer:
  rules:
    - camel_case_types
    - empty_constructor_bodies
```
If you don't list a rule it isn't checked.

**Comment by dikmax** [comment]
@zoechi I've tried that first. It doesn't work.

Then open sources for linter to understand how it works with configs: https://github.com/dart-lang/linter/blob/master/lib/src/config.dart 

If I understand everything correctly, it should read and parse `linter` root key from .yaml file provided by config option, but it doesn't.

After these steps I've created this issue.

**Comment by zoechi** [comment]
`.analysis_options` is to enable lint rule checking by the dartanalyzer (this linter is run as plugin). If you run the linter directly you need to pass a configuration file like

```yaml
# File set description: lists of patterns to include/exclude
files:
  include:
    - '**/bin/**'
    - '**/example/**'
    - '**/lib/**'
    - '**/test/**'
    - '**/tool/**'
  exclude:
    - test/.tmp_data/**

rules:
  lint:
    unnecessary_getters: false # TODO(zoechi) remove when #35 is fixed
    unnecessary_brace_in_string_interp: false
```
```sh
pub global activate linter
pub global run linter --config my_config.yaml
```

**Comment by dikmax** [comment]
That's much better, you should include that somewhere in readme!

But there's still some glitches. 

```bash
pub global run linter --config my_config.yaml
```

It won't work without some folder or file provided as command-line argument. I've just added `.` at the end. Files exclude not works as expected:

```yaml
files:
  include:
    - 'bin/**'
    - 'lib/**'
    - 'test/**'
    - 'tool/**'
  exclude:
    - test/data/**
```

Still finds lint errors in test/data/*.dart

**Comment by pq** [comment]
Thanks for the feedback @dikmax!  The `linter` commandline tool has fallen a bit behind the analyzer proper (hence the lack of support for excludes).  I'll fix that (https://github.com/dart-lang/linter/issues/149).  Clearly the docs could use some attention as well.

**Comment by pq** [comment]
OK, looks like that is actually as designed.  The rub is that the same file is excluded and included and our semantics favor the include.  Feel free to reopen if you think we should revisit.

Thanks!
